backend/src/v2/driver/driver.go

```diff
@@ -436,14 +438,42 @@ func DAG(ctx context.Context, opts Options, mlmd *metadata.Client) (execution *E
 		return execution, fmt.Errorf("ArtifactIterator is not implemented")
 	}
 	isIterator := opts.Task.GetParameterIterator() != nil && opts.IterationIndex < 0
+	// Fan out iterations
 	if execution.WillTrigger() && isIterator {
 		iterator := opts.Task.GetParameterIterator()
-		value, ok := executorInput.GetInputs().GetParameterValues()[iterator.GetItems().GetInputParameter()]
 		report := func(err error) error {
 			return fmt.Errorf("iterating on item input %q failed: %w", iterator.GetItemInput(), err)
 		}
-		if !ok {
-			return execution, report(fmt.Errorf("cannot find input parameter"))
+		// Check the items type of parameterIterator:
+		// It can be "inputParameter" or "Raw"
+		value := structpb.NewNullValue()
+		var ok bool
```

Comment:
Got it, I will declare `value` and `ok` in the same line.

Reply:
Remove the declaration for `ok`?

Reply:
The problem is, in line 454 (just 4 lines below), we need to assign values to both `value` and `ok`.

Reply:
No, you could have 
```go
var value *structpb.Value
value, ok := ...
```

Short variable declaration works in this case.

Reply:
I see! I will change that.

Reply:
It still reports error though, probably because `value, ok := ...` line is not directly under the declaration of `value`, but inside a switch/case.

Reply:
> It still reports error though, 

What error specifically? 

>  probably because `value, ok := ...` line is not directly under the declaration of value, but inside a switch/case.

I don't think this would matter.
You can probably play around here: https://go.dev/tour/basics/10

Reply:
See an example here: https://go.dev/play/p/-LdL96A5GUV
If I use the `:=` declaration inside the switch/case block, the value of `value` that was assigned inside won't be accessible from outside.

Reply:
That's expected. But we were saying to keep the `value` declaration as-is (we already settled on that one from another comment where I asked to remove the null value assignment and only keep the declaration). The comment above was to omit the `ok` declaration on this line, because I don't think you need to read `ok` outside the switch.

Reply:
Does that mean I declare `ok` inside the switch?

Reply:
you're right on the value part, just a nitpick let's still move ok declaration inside the switch so it's close to where it is used.